src/storage/mod.rs

```diff
@@ -1446,6 +1446,24 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
         Ok(())
     }
 
+    // Schedule raw modify commands, which reuse the txn scheduler.
+    // TODO: seperate the txn and raw commands if needed in the future.
+    fn sched_raw_command<T>(&self, future: T) -> Result<()>
```

Comment:
1. Suggest to check deadline exceeded in the begining of future, to avoid processing requests already timeout in client side. Refer to [here](https://github.com/tikv/tikv/blob/73c5d13455e0f07e5b38dd5de2d346afa2f1ddef/src/storage/txn/scheduler.rs#L511).

2. Suggest to add some scheduler related metrics, such as `SCHED_STAGE_COUNTER_VEC`, `SCHED_HISTOGRAM_VEC_STATIC`, to help trouble shootings (e.g., slow writes).

Reply:
Added deadline check and metric. PTAL~